lib/msal-common/src/app/module/AuthorizationCodeModule.ts

```diff
@@ -64,17 +72,75 @@ export class AuthorizationCodeModule extends AuthModule {
         // Populate query parameters (sid/login_hint/domain_hint) and any other extraQueryParameters set by the developer
         requestParameters.populateQueryParams();
 
-        return requestParameters.createNavigateUrl();
+        const urlNavigate = await requestParameters.createNavigateUrl();
+
+        // Cache token request
+        const tokenRequest: TokenExchangeParameters = {
+            scopes: requestParameters.scopes.getOriginalScopesAsArray(),
+            resource: request.resource,
+            codeVerifier: requestParameters.generatedPkce.verifier,
+            extraQueryParameters: request.extraQueryParameters,
+            authority: requestParameters.authorityInstance.canonicalAuthority,
+            correlationId: requestParameters.correlationId,            
+        };
+
+        this.cacheStorage.setItem(TemporaryCacheKeys.REQUEST_PARAMS, this.cryptoObj.base64Encode(JSON.stringify(tokenRequest)));
+
+        return urlNavigate;
     }
 
     async createAcquireTokenUrl(request: AuthenticationParameters): Promise<string> {
         throw new Error("Method not implemented.");
     }
 
+    async acquireTokenAuto(codeResponse: CodeResponse): Promise<TokenResponse> {
```

Comment:
Question: what's the use case here in comparison to the other `acquireToken` functions?

Reply:
Good question! I was kicking this around a bit, and I wasn't sure how to best construct this.

Essentially, this function is used when we want to immediately want to exchange a auth code for a token. Otherwise, we return the code response and leave it up to the user to exchange for a token by calling acquireToken themselves. My question: Do we leave this up to the user, or do we just handle both legs of the flow immediately?